MCprep_addon/materials/prep.py

```diff
@@ -35,171 +36,158 @@
-	anim_row.enabled = can_swap_text
-	if can_swap_text:
-		anim_row.prop(self, "animateTextures")
-	else:
-		anim_row.prop(self, "animateTextures", text="Animate textures (disabled due to import settings)")
-	col.prop(self, "autoFindMissingTextures")
-
-	row = self.layout.row()
-	row.prop(self, "useExtraMaps")
-	row.prop(self, "syncMaterials")
-	row = self.layout.row()
-	# row.prop(self, "newDefault")
-
-	# col = row.column()
-	# col.prop(self, "normalIntensity", slider=True)
-
-	row = self.layout.row()
-	col = row.column()
-	col.prop(self, "improveUiSettings")
-	col = row.column()
-	col.prop(self, "combineMaterials")
-	row = self.layout.row()
-	row.prop(self, "optimizeScene")
-	row.prop(self, "useEmission")
-
-
-class MCPREP_OT_prep_materials(bpy.types.Operator, McprepMaterialProps):
-	"""Fixes materials and textures on selected objects for Minecraft rendering"""
-	bl_idname = "mcprep.prep_materials"
-	bl_label = "MCprep Materials"
-	bl_options = {'REGISTER', 'UNDO'}
-
-	skipUsage: bpy.props.BoolProperty(
-		default=False,
-		options={'HIDDEN'})
-
-	def invoke(self, context, event):
-		return context.window_manager.invoke_props_dialog(
-			self, width=300 * util.ui_scale())
-
-	def draw(self, context):
-		draw_mats_common(self, context)
-
-	track_function = "materials"
-	track_param = None
-	track_exporter = None
-	@tracking.report_error
-	def execute(self, context):
-
+	def draw_before(self, context: Context) -> None:
+		pass
+	
+	def draw_after(self, context: Context) -> None:
+		pass
+
+	def draw(self, context: Context) -> None:
+		self.draw_before(context)
+		# For swap texture if prepMaterials is on
+		if hasattr(self, "prepMaterials"):
+			if not getattr(self, "prepMaterials"):
+				return 
+		row = self.layout.row()
+		col = row.column()
+		engine = context.scene.render.engine
+		if engine == 'CYCLES' or engine == 'BLENDER_EEVEE':
+			col.prop(self, "packFormat")
+			col.prop(self, "usePrincipledShader")
+		col.prop(self, "useReflections")
+		col.prop(self, "makeSolid")
+	
+		anim_row = col.row()
+		can_swap_text = util.is_atlas_export(context)
+		anim_row.enabled = can_swap_text
+		if can_swap_text:
+			anim_row.prop(self, "animateTextures")
+		else:
+			anim_row.prop(self, "animateTextures", text="Animate textures (disabled due to import settings)")
+		col.prop(self, "autoFindMissingTextures")
+	
+		row = self.layout.row()
+		row.prop(self, "useExtraMaps")
+		row.prop(self, "syncMaterials")
+		row = self.layout.row()
+		# row.prop(self, "newDefault")
+	
+		# col = row.column()
+		# col.prop(self, "normalIntensity", slider=True)
+	
+		row = self.layout.row()
+		col = row.column()
+		col.prop(self, "improveUiSettings")
+		col = row.column()
+		col.prop(self, "combineMaterials")
+		row = self.layout.row()
+		row.prop(self, "optimizeScene")
+		row.prop(self, "useEmission")
+		
+		self.draw_after(context)
+		
+	def prep_mat(self, context: Context):
+		"""A method for prepping material
+		
+		Returns: 0 Success, 1 Error
+		"""
```

Comment:
Don't use integers for success or failure, use booleans. This is Python, not C.

Reply:
I'm less concerned about integers vs otherwise, but if we do want to understand the result of the operation, I think an annotated function return of `) -> bool:` would do nicely.